src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts

```diff
@@ -1,21 +1,231 @@
+    const getVendorByIdSpy = jest.spyOn(vendorService, 'getVendorById').mockImplementation(() => of(getVendorData()));
+    const addVendorSpy = jest.spyOn(vendorService, 'addVendor').mockImplementation(() => of(returnResponse()));
+    const updateVendorSpy = jest.spyOn(vendorService, 'updateVendor').mockImplementation(() => of(returnResponse()));
+    const routerSpy = jest.spyOn(router, 'navigate').mockImplementation(() => Promise.resolve(true));
+    const messageServiceSpy = jest.spyOn(messageService, 'add').mockImplementation(() => {});
+    return {
+      getMarketsSpy,
+      getServicesSpy,
+      getVendorByIdSpy,
+      addVendorSpy,
+      updateVendorSpy,
+      routerSpy,
+      messageServiceSpy,
+    };
+  };
+  let urlValue = { url: 'vendor/creation', navigate: jest.fn() };
+
   it('should create', () => {
+    const { component } = setup();
     expect(component).toBeTruthy();
   });
+
+  it('should create vendor form when component initializes', () => {
+    const { component } = setup();
+    component.ngOnInit();
+    expect(component.addVendorForm).toBeDefined();
+  });
+
+  it('should populate data when user open edit screen', () => {
+    const { component } = setup();
+    urlValue.url = 'vendor/edit/1';
+    component.ngOnInit();
+    component.fetchVendorData();
+    expect(component.addVendorForm.get('vendorName')?.value).toBe('Vendor 1');
+  });
+
+  it('should update the vendor data when update form is submitted', () => {
+    const { component, spyObj } = setup();
+    component.isEdit = true;
+    component.selectedVendorId = 1;
+    component.addVendorForm = getVendorServiceFormMock();
+    component.submitVendor();
+    expect(spyObj.updateVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should show error during updating the vendor data', () => {
+    const { component, spyObj } = setup();
+    component.isEdit = true;
+    component.selectedVendorId = 1;
+    component.addVendorForm = getVendorServiceFormMock();
+    const errorResponse = {
+      error: {
+        errors: {
+          error: [
+            {
+              message: '',
+            },
+          ],
+        },
+        message: 'Error occurred',
+      },
+    };
+    spyObj.updateVendorSpy.mockReturnValue(throwError(() => errorResponse));
+    component.submitVendor();
+    expect(spyObj.updateVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should add the vendor data when form is submitted', () => {
+    const { component, spyObj } = setup();
+    component.isEdit = false;
+    component.addVendorForm = getVendorServiceFormMock();
+    component.submitVendor();
+    expect(spyObj.addVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should display error when they are present during vendor form submission.', () => {
+    const { component, spyObj } = setup();
+    component.addVendorForm = getVendorServiceFormMock();
+    const errorResponse = {
```

Comment:
make this as method and reuse this error response

Reply:
done